### PR TITLE
permissions: modify error handling(for leaving team) and permissions(for joining team)

### PR DESCRIPTION
**Pull request by m-e-l-u-h-a-n**
This commit adds permission for joining a team only on complete profile
and modifies member exit serializer to send proper response when hackthon
is not found.

### CODE REVIEW

File: core/views.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ class TeamView(generics.RetrieveUpdateDestroyAPIView):
     """
 
     def get_permissions(self):
-        if self.request.method == "GET":
+        if self.request.method in ["GET"]:
```
Comment (shubhanshu02) [inline core/views.py]: Instead of simply using `GET`, using `SAFE_METHODS` would be a better choice.

**Review by shubhanshu02 — APPROVED** [review]
LGTM. Just have a look at the below comment.

File: core/views.py (hunk 1 above)
Comment (m-e-l-u-h-a-n) [reply core/views.py]: That change was  commited by mistake. For now let it be explicitly clear about `GET` method only.

**Review by aishwary023 — APPROVED** [review]
LGTM!

**Pull request merged**